### PR TITLE
fcm make: build: flexible checksum method

### PR DESCRIPTION
**Pull request by matthewrmshin**
MD5 or one of SHA algorithms in Digest::SHA.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@benfitzpatrick please review.

File: lib/FCM/Util.pm [hunk 1]
```diff
@@ -695,7 +711,7 @@ Returns a file handle for loading contents from $path.
 
 =item $u->file_md5($path)
 
-Returns the MD5 checksum of $path.
+Deprecated. Equivalent to $u->file_checksum($path, 'md5').
```
Comment (benfitzpatrick) [inline lib/FCM/Util.pm]: Why deprecate it? Why not delete it?

Comment (matthewrmshin) [inline lib/FCM/Util.pm]: Just in case something else like GEN is still using this library.

File: t/fcm-make/48-build-sha1.t [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+TEST_KEY="${TEST_KEY_BASE}-incr"
+run_pass "${TEST_KEY}" fcm make
+run_pass "${TEST_KEY}-sha1sum-check" sha1sum -c - <<<"$(get-hello-checksum)"
+HELLO_O_MTIME_INCR="$(stat -c '%Y' 'build/o/hello.o')"
+run_pass "${TEST_KEY}-mtime-of-hello.o" \
+    test "${HELLO_O_MTIME_INCR}" -eq "${HELLO_O_MTIME}"
+#-------------------------------------------------------------------------------
+TEST_KEY="${TEST_KEY_BASE}-incr-2"
+sed -i 's/Hello/Greet/' 'src/hello.f90'
+sha1sum 'src/hello.f90' >'hello.f90.sha1sum'
```
Comment (benfitzpatrick) [inline t/fcm-make/48-build-sha1.t]: What does this do?

Comment (matthewrmshin) [inline t/fcm-make/48-build-sha1.t]: Line removed. (It was used for visualization while the test was developed.)

**Comment by benfitzpatrick** [comment]
OK, looks fine. Running the test-battery...

**Pull request merged**